samples/cpp/hello_world/README.rst

```diff
@@ -0,0 +1,33 @@
+.. _hello_cpp_world:
+
+Hello C++ World
+###############
+
+Overview
+********
+
+A simple C++ sample that can be used with all :ref:`supported board <boards>` and prints
```

Comment:
> Well, it does say "all supported boards", so a simple interpretation of that would be to say that unsupported boards are unsupported.

but it links to the full list of Zephyr supported boards, so I am not sure how one -- especially people who are the audience for such a Hello World sample, i.e more than likely Zephyr beginners -- would be supposed to know that there are, in fact, unsupported boards?

Reply:
I created #66027 to track the xtensa issue and zephyrproject-rtos/sdk-ng#593 actually existed from a long time ago when I tried to create a simple hello_world app too. I will make a note in those tickets to adjust the `sample.yaml` file when the fix is complete.

Reply:
Personally though, I would rather not revisit this application since it's so trivial, and I don't think we need to include bug reports (similarly to cmake internals and platform specific workarounds) in something as basic as hello world.

Reply:
To clarify, I was asking for improving the wording in the README, not the sample.yaml which I would assume a vast majority of the audience for "Hello World" kind of samples don't even know exist. If I am getting started with Zephyr and giving the C++ Hello World a try, there is a very high chance I will only look at the README initially, so if it says "this sample can be used on all supported boards", it is really confusing. And some people might not even read the README the at all, but that's where they will probably refer to in case it doesn't work, to check what they may have done wrong.

FWIW I agree that hopefully things will be fixed by the time we cut a release, but in the meantime (or in the event that it does *not* get fixed before the release), I really think the documentation should point out any potential catch or limitation.